pkg/utils/publicip/detector_test.go

```diff
@@ -81,9 +81,10 @@ var _ = Describe("IpifyDetector", func() {
 		})
 
 		It("should return only the available IP", func() {
-			Expect(detector.DetectPublicIPs(ctx, log)).Error().To(MatchError(
+			Expect(detector.DetectPublicIPs(ctx, log)).Error().To(MatchError(Or(
```

Comment:
I was looking into this flake without noticing the flake issue or your PR.
I suggest stabilizing the order of API probing (first IPv4, then IPv6) instead of augmenting the test matchers. You can have a look at https://github.com/gardener/gardener/pull/12492 and decide what approach we should take.